capstone/src/main/java/com/google/univiz/scorecard/CollegeIdReaderProviderImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.google.univiz.scorecard;
+
+import com.google.univiz.api.CollegeId;
+import com.google.univiz.config.UnivizConfig;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import javax.inject.Inject;
+
+class CollegeIdReaderProviderImpl implements CollegeIdReaderProvider {
+  private final UnivizConfig univizConfig;
+
+  @Inject
+  public CollegeIdReaderProviderImpl(UnivizConfig univizConfig) {
+    this.univizConfig = univizConfig;
+  }
+
+  @Override
+  public Reader getReaderFromCollegeId(CollegeId id) {
```

Comment:
FWIW - I think this class should be lightweight as possible - including the URL generation (that way we can verify that the URL is generated correctly in test):

interface CollegeIdReaderProvider {
  InputStream getStreamFromUrl(String URL) throws IOException;
}

Plus if we use a strategy like this we can mock the exceptional cases...

Reply:
Ah yes, that will be useful, will do.